TEP003_commit_message_guidelines.rst

```diff
@@ -0,0 +1,119 @@
+TEP003: Git Policy
+========================
+
+Status
+======
+
+ **Discussion**
+
+Responsible
+===========
+
+@yeganer, @orbitfold
+
+Description
+===========
+
+Commit Message Guidelines
+-------------------------
+
+When committing changes do not use -m option. This is only suitable for really short commit messages. Instead omit the message flag and you will be taken to your default text editor where you can enter the message. If you have already pushed the changes to a remote repository or if you have committed a change locally with a message that does not conform to the guidelines given below please use the following instructions to change commit messages:
```

Comment:
I think we should give basic examples for `git commit --amend` and `git rebase upstream/master` possible mention `git rebase -i`, too.

I think we should emphasize to never use `git commit -m`. Even one liners should be written in the Editor. Since then one perhaps thinks of something to add and you end up with a 3 line commit.